sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.cs

```diff
@@ -159,19 +169,32 @@ private void OnPost(object state)
 
                 var dataPoint = new MonitoringDataPoint
                 {
-                    Version = "DotNetLiveMetricsPOC", // sdk version
-                    MachineName = "Desktop-Name",
-                    Instance = "Desktop-Name",
+                    Version = SdkVersionUtils.s_sdkVersion.Truncate(SchemaConstants.Tags_AiInternalSdkVersion_MaxLength),
                     InvariantVersion = 5,
-                    Timestamp = DateTime.UtcNow,
-                    PerformanceCollectionSupported = false,
-                    IsWebApp = false,
-                    RoleName = null,
+                    Instance = liveMetricsResource?.RoleInstance ?? "UNKNOWN_INSTANCE",
+                    RoleName = liveMetricsResource?.RoleName,
+                    MachineName = Environment.MachineName, // TODO: MOVE TO PLATFORM
+                    // TODO: Is the Stream ID expected to be unique per post? Testing suggests this is not necessary.
                     StreamId = _streamId,
+                    Timestamp = DateTime.UtcNow,
+                    //TODO: Provide feedback to service team to get this removed, it not a part of AI SDK.
                     TransmissionTime = DateTime.UtcNow,
+                    IsWebApp = IsWebAppRunningInAzure(),
```

Comment:
Not from your PR - this method gets called every time, we should consider caching it.